cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -783,6 +784,35 @@ Status NumPyConverter::ConvertDecimals() {
   return PushBuilderResult(&builder);
 }
 
+Status NumPyConverter::ConvertDateTimes() {
+  // Convert array of datetime.datetime objects to Arrow
+  PyAcquireGIL lock;
+  PyDateTime_IMPORT;
+
+  Ndarray1DIndexer<PyObject*> objects(arr_);
+
+  // datetime.datetime stores microsecond resolution
+  Time64Builder builder(::arrow::timestamp(TimeUnit::MICRO), pool_);
```

Comment:
This should be `TimestampBuilder`?

Reply:
Yes, this is a copy&paste artefact. Probably some code duplication we may eliminate in the future.